rfcs/convergence/file-naming.md

```diff
@@ -0,0 +1,149 @@
+| useExample.ts       | example.hook.ts     | `export const useExample = () => {}`                                                | main logic encapsulated under hook -> name should be the same as encapsulated component (in our case `example`)                                                                                                                                                                                                           |
+| useExampleStyles.ts | example.styles.ts   | `export const useStyles = makeStyles(); export const useIconStyles = makeStyles();` | with `makeStyles` 1 to N hooks for particular component styling needs. Why we currently use a solution that provides hooks, having `.styles` could be a good enough convention to explicitly communicate the intent. (if we switch implementation in the future we wouldn't need to change the type name as it's generic) |
+| Example.types.ts    | example.types.ts    | `export type Foo = {}; export interface Hello {}`                                   | aggregated set of reusable type definitions used within other parts of component/also might be re-exported for public API surface                                                                                                                                                                                         |
+
+### Explanation
+
+- use consistent names for all files.
+- follow a pattern that describes the symbol's feature or the file scope(component name) then its type.
+  - The recommended pattern is `feature.type.ts`.
+- use conventional **type** names including
+  - `.hook`
+  - `.view`
+  - `.styles`
+  - `.types`
+  - `.stories`
+  - `.utils`
+  - invent additional type names if you must but take care not to create too many
+
+**Why?**
+
+- Type names provide a consistent way to quickly identify what is in the file.
+- Type names make it easy to find a specific file type using an editor or IDE's fuzzy search techniques.
+- Type names provide pattern matching for any automated tasks.
+
+---
+
+- use `.tsx` only for files that contain view/rendering related logic
+
+**Why?**
+
+- will enforce to use JSX only where necessary (renderer only)
+- same as for Type Names
+
+---
+
+- (optional) use `.spec.` instead `.test.` suffix as we already follow behavioral driven approach (describes/it) rather than atomic `test()` approach
+  - as a next step, enable lint rule that will enforce one way or another within tests
+
+**Why?**
+
+- consistency/proper intent
+
+---
+
+- (optional) use kebab case for naming
+
+**Why?**
+
+- kebab case is more human readable especially in longer words
+- mitigates issues with git on various OS/case in/consistent files systems (eg. rename `helloWorld` to `helloworld`, commit, push, merge... now someone wants to change it back to `helloWorld`... )
```

Comment:
This is handled by TypeScript compiler via `forceConsistentCasingInFileNames` (https://www.typescriptlang.org/docs/handbook/compiler-options.html) and is enabled across the repo. Is it a real currently?